src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java

```diff
@@ -50,115 +44,130 @@ public String getUrlForPage(String wikiName, String pageName) {
-  }
-
   public String getUrlForWiki(String wikiName, boolean addWWW) {
     return getUrlForWiki(wikiName, addWWW, envType);
   }
 
 
   public String getUrlForWiki(String wikiName, boolean addWWW, EnvType envType) {
+    return getUrlForWiki(wikiName, DEFAULT_LANGUAGE, addWWW, envType);
+  }
+
+  public String getUrlForWiki(String wikiName, String language) {
+    return getUrlForWiki(wikiName, language, false, envType);
+  }
+
+  public String getUrlForWiki(String wikiName, String language, boolean addWWW, EnvType envType) {
     final String wikiaName = getWikiaGlobalName(wikiName);
 
     String www = "";
     if (addWWW) {
       www = "www.";
     }
 
+    String host;
+    HttpUrl.Builder urlBuilder = new HttpUrl.Builder();
     switch (envType) {
       case DEV: {
         String devBoxOwner = this.env.split("-")[1];
-        return String.format(getUrlProtocol() + DEV_URL_FORMAT, www, wikiaName, devBoxOwner, envType.getWikiaDomain());
+        host = String.join(".", www + wikiaName, devBoxOwner, envType.getWikiaDomain());
+        break;
       }
       case PROD: {
-        return String.format(getUrlProtocol() + PROD_URL_FORMAT, www, wikiaName, envType.getWikiaDomain());
+        host = String.join(".", www + wikiaName, envType.getWikiaDomain());
+        break;
       }
       case STAGING: {
-        return String.format(getUrlProtocol() + PROD_URL_FORMAT, www, wikiaName, envType.getWikiaDomain());
+        host = String.join(".", www + wikiaName, envType.getWikiaDomain());
+        break;
       }
       case SANDBOX: {
-        if (newStagingUrlFormat) {
-          return String.format(getUrlProtocol() + SANDBOX_NEW_URL_FORMAT, www, wikiaName, this.env, envType.getWikiaDomain());
-        } else {
-          return String.format(getUrlProtocol() + SANDBOX_URL_FORMAT, this.env, www, wikiaName, envType.getWikiaDomain());
-        }
+        host = String.join(".", www + wikiaName, this.env, envType.getWikiaDomain());
+        break;
       }
       default:
-        return "";
+        throw new WebDriverException("Unknown environment type");
+    }
+    urlBuilder.scheme(getUrlProtocol()).host(host);
+
+    if (language != null && !language.equals("en")) {
+      urlBuilder.addPathSegments(language);
     }
+    return urlBuilder.build().toString();
   }
 
   public String getUrlProtocol() {
-    if (this.forceHttps) {
-      return HTTPS_PREFIX;
-    }
-    return HTTP_PREFIX;
+    return this.forceHttps ? "https" : "http";
   }
 
   public String getWikiGlobalURL() {
+    String host;
+    HttpUrl.Builder urlBuilder = new HttpUrl.Builder();
     EnvType env = Configuration.getEnvType(this.env);
 
     switch (env) {
       case DEV: {
         String devBoxOwner = this.env.split("-")[1];
-        return String.format(getUrlProtocol() + "www.%s.%s", devBoxOwner, env.getWikiaDomain());
+        host = String.join(".", "www", devBoxOwner, envType.getWikiaDomain());
+        break;
       }
       case SANDBOX: {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested code block into a method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1199)